src/wallet/txbuilder.h

```diff
@@ -36,10 +36,10 @@ class TxBuilder
     explicit TxBuilder(CWallet& wallet) noexcept;
     virtual ~TxBuilder();
 
-    CWalletTx Build(const std::vector<CRecipient>& recipients, CAmount& fee);
+    CWalletTx Build(const std::vector<CRecipient>& recipients, CAmount& fee, const CCoinControl *coinControl = NULL);
```

Comment:
This will cause more dependencies at the abstraction level. Also cause more requirements to the derived class. We should pass `CCoinControl` to constructor of `SigmaSpendBuilder` instead.

Reply:
Sounds good.